napi/src/options.rs

```diff
@@ -141,6 +141,10 @@ pub struct NapiResolveOptions {
     ///
     /// Default `false`
     pub builtin_modules: Option<bool>,
+
+    /// When enabled, will attempt to reduce memory usage by clearing data and fields
+    /// that are not relevant to the current process, and shuold not be held in memory.
+    pub reduce_memory_usage: Option<bool>,
```

Comment:
I found myself needing some of the fields on the package.json data, so instead of always removing it, I wrapped it in an option.